### PR TITLE
Add support for custom social-sharing images

### PR DESCRIPTION
**Pull request by stevejalim**
Added via the `wagtailmetadata` package (BSD licence), which gives us OG and sharing tags for FB, Twitter, etc via a simple templatetag and an additonal model field

We can't add the templatetag to the base template, because it will cause pages like 404.html to blwo up, because they don't have `self` in their context. So, instead, the tag has to be called on actual page templates, not a base.

Example of new image field in the UI

Screenshot 2023-12-12 at 22 49 26

Example output:

Screenshot 2023-12-12 at 22 49 40

### To test locally:

* `just export-local-data` to back up your current local state
* check out branch
* ~`just migrate`~ `just preflight`
* Go to CMS admin and edit a page - on the Settings tab you'll see the new option to add a social image.
* Publish and view source to confirm you're getting a resized image as your sharing image

### CODE REVIEW

**Comment by stevejalim** [comment]
@craigcook Wanna give this a try locally? Once it's approved we can get it up on Dev. I can merge it tomorrow and deploy it to futuremo-dev

File: birdbox/birdbox/settings/base.py [hunk 1]
```diff
@@ -347,6 +348,9 @@ def path_from_root(*args):
 
 ENGAGE_ROBOTS = config("ENGAGE_ROBOTS", parser=bool, default="False")
 
+# Customise the size of the social-sharing card image
+WAGTAILMETADATA_IMAGE_FILTER = "fill-1200x630"
```
Comment (craigcook) [inline birdbox/birdbox/settings/base.py:352]: I think this gets appended to the filename even when it's less than 1200 x 630 and no resizing occurs. It doesn't actually hurt anything but might make it confusing when the implied dimensions don't match the actual dimensions. Maybe this should be a more generic suffix like "-og" or something?

**Review by craigcook — APPROVED** [review]
Does what it says on the tin. r+, and if you want to change the image suffix that's up to you.

File: birdbox/birdbox/settings/base.py (hunk 1 above)
Comment (stevejalim) [reply birdbox/birdbox/settings/base.py:352]: Yeah, that's the unchangeable naming convention of Wagtail. I get your point, but I think the risk of confusion is tolerable.

I'm not a huge fan of how the social sharing images have to get cropped to a letterbox (as per Meta's social-sharing recommendations etc) so we'll have to watch out for people uploading square images that get poorly auto-cropped and people aren't aware until too late

**Pull request merged**